File: solutions/0165-compare-version-numbers/solution.js
/**
 * @param {string} version1
 * @param {string} version2
 * @return {number}
 */
var compareVersion = function(version1, version2) {
    version1 = version1.split(".").map(el => parseInt(el))
    version2 = version2.split(".").map(el => parseInt(el))
    
    const l = Math.max(version1.length, version2.length);
    
    for(let i = 0; i < l; i++){
        const v1 = version1[i] ?? 0;
        const v2 = version2[i] ?? 0;
        
        
        if(v1 < v2){
            return -1
        }else if(v2 < v1){
            return 1
        }
    }
    
    return 0
};

File: solutions/0165-compare-version-numbers/solution.py
class Solution:
    def compareVersion(self, version1: str, version2: str) -> int:
        version1 = version1.split(".")
        version2 = version2.split(".")
        
        v1l = len(version1)
        v2l = len(version2)
        max_l = max(v1l, v2l)

        for i in range(0, max_l):
            if i >= v1l:
                v1= 0
            else:
                v1 = int(version1[i])
            
            if i >= v2l:
                v2= 0
            else:
                v2 = int(version2[i])
            
            if v1 > v2:
                return 1
            elif v2 > v1:
                return -1
        
        return 0
